### PR TITLE
Add images to each profile

### PR DESCRIPTION
**Pull request by timcleaver**
Added the ability to include images in addition to colors for each switch role link. The image is also included in the header bar of the console.  Images are set by adding image = <url> to each profile config entry. It does mean that each image must be either served by a local web server or are otherwise online.

### CODE REVIEW

**Comment by tilfin** [comment]
@timcleaver Tell me about your usecase, and where do you place the images actually?

**Comment by timcleaver** [comment]
I work at a consultancy and it is easier for me to recognize the switch role accounts by the clients' logos than remember them by color (which can become problematic if two clients share similar color palettes). The images are placed where the color patch is/was displayed per account and similarly to the left of the selected role/account in the top menu bar.

**Comment by tilfin** [comment]
@timcleaver Thank you, I'm sorry that my last question is for image **file** storage. Do you place their image files on S3?

**Comment by timcleaver** [comment]
No, there is no image storage. The user supplies the uri of an image in the config file for each role the want an image for. The uri is injected into image tags (i.e ). The user can self host the images using a local web server, use S3, or find uris from google, or provide any uri that will correctly serve the image over HTTP(S).

File: src/lib/content.js [hunk 1]
```diff
@@ -122,20 +122,38 @@ function loadProfiles(profile, list, csrf, hidesHistory, hidesAccountId) {
...
+         <form action="https://signin.aws.amazon.com/switchrole" method="POST" target="_top" data-aesr-profile="${item.profile}">
+          <input type="hidden" name="action" value="switchFromBasis">
+          <input type="hidden" name="src" value="nav">
+          <input type="hidden" name="roleName" value="${item.role_name}">
+          <input type="hidden" name="account" value="${item.aws_account_id}">
+          <input type="hidden" name="mfaNeeded" value="0">
+          <input type="hidden" name="color" value="${color}">
+          <input type="hidden" name="csrf" value="${csrf}">
+          <input type="hidden" name="redirect_uri" value="${redirectUri}">
+          <label for="awsc-recent-role-switch-0" class="awsc-role-color"><img src=${item.image.replace(/"/g, '')} width="100%" height="100%"></label>
```
Comment (tilfin) [inline src/lib/content.js]: Specify width and height by **em** unit

File: src/lib/content.js [hunk 2]
```diff
@@ -200,9 +218,13 @@ function attachColorLine(profiles) {
     const found = profiles.find(item => { return item.profile === profileName });
     const color = found && found.color || null;
 
+    var label = usernameMenu.querySelector('.nav-elt-label');
+    if (found && found.image) {
+        label.insertAdjacentHTML('beforebegin', Sanitizer.escapeHTML(`<img id="AESW_Image" src=${found.image.replace(/"/g, '')} width="17em" height="17em" style="float: left; padding-right: 1em">`));
```
Comment (tilfin) [inline src/lib/content.js]: I guess that _17em_ is too big. I would like to see this screenshot.

**Comment by timcleaver** [comment]
Here is a screenshot after using fixed ems for the image sizes:
screenshot-edited

File: src/lib/content.js [hunk 3]
```diff
@@ -122,20 +122,38 @@ function loadProfiles(profile, list, csrf, hidesHistory, hidesAccountId) {
...
+         <form action="https://signin.aws.amazon.com/switchrole" method="POST" target="_top" data-aesr-profile="${item.profile}">
+          <input type="hidden" name="action" value="switchFromBasis">
+          <input type="hidden" name="src" value="nav">
+          <input type="hidden" name="roleName" value="${item.role_name}">
+          <input type="hidden" name="account" value="${item.aws_account_id}">
+          <input type="hidden" name="mfaNeeded" value="0">
+          <input type="hidden" name="color" value="${color}">
+          <input type="hidden" name="csrf" value="${csrf}">
+          <input type="hidden" name="redirect_uri" value="${redirectUri}">
+          <label for="awsc-recent-role-switch-0" class="awsc-role-color"><img src=${item.image.replace(/"/g, '')} width="14em" height="14em"></label>
```
Comment (tilfin) [inline src/lib/content.js]: The width and height attributes of img support only **px** as unit, so add the style attribute like `style="width: 1em; height: 1em"` to it.

File: src/lib/content.js [hunk 4]
```diff
@@ -200,9 +218,13 @@ function attachColorLine(profiles) {
     const found = profiles.find(item => { return item.profile === profileName });
     const color = found && found.color || null;
 
+    var label = usernameMenu.querySelector('.nav-elt-label');
+    if (found && found.image) {
+        label.insertAdjacentHTML('beforebegin', Sanitizer.escapeHTML(`<img id="AESW_Image" src=${found.image.replace(/"/g, '')} width="14em" height="14em" style="float: left; padding-right: 1em">`));
```
Comment (tilfin) [inline src/lib/content.js]: The width and height attributes of img support only **px** as unit, so add the style attribute like `style="width: 1em; height: 1em"` to it.

**Comment by tilfin** [comment]
@timcleaver LGTM! but please add only the `image` parameter explanation to https://github.com/tilfin/aws-extend-switch-roles/blob/master/README.md#optional-parameters

**Comment by tilfin** [comment]
@timcleaver Thank you!

**Pull request merged**